raiden-ts/docs-source/SDK-Development.md

```diff
@@ -193,22 +194,22 @@ Reducers are one of the simplest parts to write. Just some heads-up:
 - Try to always be as type strict as possible. lodash's `get` and `set` usually aren't typesafe (because of their nested sweetness), so add explicit declarations and narrowing checks whenver needed, to be warned when changing anything that could break a reducer.
 - Don't overthink reducers: you can always assume you're receiving a valid/consistent state and action, so no need to add too much checks besides the typing ones, just be sure to always return a valid/consistent state as well and we should be good.
 - Each `action` type must be handled only once, in a single and very specific place, to avoid issues with the `reduce-reducers` pattern used in the root reducer declaration.
-- The state must be JSON-serializable. Try to stay on the primitive types as much as possible, unless there's a good reason to have an instance there, and it must be serializable (with custom `io-ts` codecs, like `BigNumber`s).
+- The state must be JSON-serializable. Try to stay on the primitive types as much as possible, unless there's a good reason to have an instance there, and it must be serializable (with custom `io-ts` codecs, like `BigNumber`).
 
 ## Epics
 
 Epics can choose to act on any action or state change, or even dependencies, but it's important they try to follow the UNIX philosophy: do one thing, and do it well. So, as much as possible, each epic should listen a single event type, and output only the minimal action set as outcome of its specific task. The `action$` input observable should be declared as `Observable<RaidenAction>`, as every action goes through (although filtered as early as possible), but try to declare the output as specifically as possible, with a tagged union of any possible action output type.
 
-### Hints, tips & tricks and pits to keep an eye on developing epics:
+### Hints, tips & tricks and pitfalls when developing epics
 
 - Be careful to not complete the output observable if the system is still running, or its function will be lost, as subscriptions aren't restarted.
-- Any unhandled exception will cause a `raidenShutdown` action with the exception to go through as last action, all epics being unsubscribed, action$ and state$ completing and the whole system to shutdown. Handle any expected exception and output a respective `fail` action if relevant, or logging it and swallowing it with `EMPTY` or `ignoreElements`. Unhandled exceptions shouldn't happen in normal operation.
+- Any unhandled exception will cause a `raidenShutdown` action with the exception to go through as last action, all epics being unsubscribed, `action$` and `state$` completing and the whole system to shutdown. Handle any expected exception and output a respective `fail` action if relevant, or logging it and swallowing it with `EMPTY` or `ignoreElements`. Unhandled exceptions shouldn't happen in normal operation.
```

Comment:
The epics aren't subscribed anymore (if possible), and shutdown may not be the last action. `raidenShutdown` still triggers the input (`action$`, `state$`) to be completed, and then the individual epics have up to `httpTimeout` to detect this completion and gracefully complete on their own once they finish their latest async tasks/teardown, during which some output action may still go through and change state (although won't be received by other epics, since the input is already complete). After this timeout, if some epic didn't complete, they're logged and then unsubscribed. Only after that, we wait for the last db flush, close it, and then complete the output and `deps.latest$` and children.